flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/StateAssignmentOperation.java

```diff
@@ -63,164 +63,170 @@ public StateAssignmentOperation(
 	public boolean assignStates() throws Exception {
 
 		for (Map.Entry<JobVertexID, TaskState> taskGroupStateEntry : latest.getTaskStates().entrySet()) {
+
 			TaskState taskState = taskGroupStateEntry.getValue();
 			ExecutionJobVertex executionJobVertex = tasks.get(taskGroupStateEntry.getKey());
 
-			if (executionJobVertex != null) {
-				// check that the number of key groups have not changed
-				if (taskState.getMaxParallelism() != executionJobVertex.getMaxParallelism()) {
-					throw new IllegalStateException("The maximum parallelism (" +
-							taskState.getMaxParallelism() + ") with which the latest " +
-							"checkpoint of the execution job vertex " + executionJobVertex +
-							" has been taken and the current maximum parallelism (" +
-							executionJobVertex.getMaxParallelism() + ") changed. This " +
-							"is currently not supported.");
+			if (executionJobVertex == null) {
```

Comment:
There is no change to the code, right? Just changing the three-way if to a two-way if with a `continue`?